marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -1249,3 +1258,76 @@ instance Arbitrary TransactionOutput where
+      , NonPositiveBalance <$> arbitrary <*> arbitrary
+      , DuplicateAccount <$> arbitrary <*> arbitrary
+      , DuplicateChoice <$> arbitrary
+      , DuplicateBoundValue <$> arbitrary
+      , MaximumValueMayExceedProtocol . fromInteger <$> arbitraryPositiveInteger
+      , TransactionSizeMayExceedProtocol <$> arbitrary <*> fmap fromInteger arbitraryPositiveInteger
+      , TransactionCostMayExceedProtocol
+          <$> arbitrary
+          <*> (ExBudget <$> fmap fromInteger arbitraryPositiveInteger <*> fmap fromInteger arbitraryPositiveInteger)
+      , TransactionValidationError <$> arbitrary <*> arbitrary
+      , TransactionWarning <$> arbitrary <*> arbitrary
+      , MissingContinuation . DatumHash . toBuiltin . pack <$> replicateM 32 arbitrary
+      , pure InconsistentNetworks
+      , pure WrongNetwork
+      , IllegalAddress <$> arbitrary
+      , pure SafetyAnalysisTimeout
+      ]
+  shrink MissingRolesCurrency = mempty
+  shrink ContractHasNoRoles = mempty
+  shrink (MissingRoleToken tokenName) = MissingRoleToken <$> shrink tokenName
+  shrink (ExtraRoleToken tokenName) = ExtraRoleToken <$> shrink tokenName
+  shrink (RoleNameTooLong tokenName) = RoleNameTooLong <$> shrink tokenName
+  shrink (InvalidCurrencySymbol _) = mempty
+  shrink (TokenNameTooLong tokenName) = TokenNameTooLong <$> shrink tokenName
+  shrink (InvalidToken token) = InvalidToken <$> shrink token
+  shrink (NonPositiveBalance accountId token) =
+    (NonPositiveBalance accountId <$> shrink token)
+      <> (flip NonPositiveBalance token <$> shrink accountId)
+  shrink (DuplicateAccount accountId token) =
+     (DuplicateAccount accountId <$> shrink token)
+     <> (flip DuplicateAccount token <$> shrink accountId)
+  shrink (DuplicateChoice choiceId) = DuplicateChoice <$> shrink choiceId
+  shrink (DuplicateBoundValue valueId) = DuplicateBoundValue <$> shrink valueId
+  shrink (MaximumValueMayExceedProtocol _) = mempty
+  shrink (TransactionSizeMayExceedProtocol transaction natural) =
+    flip TransactionSizeMayExceedProtocol natural <$> shrink transaction
+  shrink (TransactionCostMayExceedProtocol transaction exBudget) =
+    flip TransactionCostMayExceedProtocol exBudget <$> shrink transaction
+  shrink (TransactionValidationError transaction string) =
+    flip TransactionValidationError string <$> shrink transaction
+  shrink (TransactionWarning transaction warning) =
+    flip TransactionWarning warning <$> shrink transaction
+  shrink (MissingContinuation _) = mempty
+  shrink InconsistentNetworks = mempty
+  shrink WrongNetwork = mempty
+  shrink (IllegalAddress address) = IllegalAddress <$> shrink address
+  shrink SafetyAnalysisTimeout = mempty
+
+
+instance Arbitrary Transaction where
+  arbitrary =
+     do
+       context <- arbitrary
+       txState <- semiArbitrary context
+       txContract <- semiArbitrary context
+       txInput <- semiArbitrary context
+       txOutput <- arbitrary
+       pure Transaction{..}
+  shrink _ = mempty
```

Comment:
Shouldn't we shrink this?

Reply:
I was undecided, but it is harmless to shrink this. Fixed in 34f1d7e5d89e68996254f165386a3264372c3252.